### PR TITLE
Stories rework: VaDropdown

### PR DESCRIPTION
**Pull request by Fsss126**
See #3676.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/.storybook/storybook-main.scss [hunk 1]
```diff
@@ -6,4 +6,5 @@
 //@import "../components/vuestic-sass/global/normalize.scss";
 @import './assets/icons/icons.scss';
 @import './assets/fonts.scss';
+@import './assets/global';
```
Comment (m0ksem) [inline packages/ui/.storybook/storybook-main.scss:9]: Can we prefer tailwind? We will forget about this approach and external contributors will use a different way.

@asvae, correct me if we're not allowed to use tailwind in stories.

File: packages/ui/src/components/va-aspect-ratio/VaAspectRatio.stories.ts [hunk 2]
```diff
@@ -9,8 +9,8 @@ export default {
 export const Default = () => ({
   components: { VaAspectRatio },
   template: `
-    <va-aspect-ratio 
-      :style="{ width: '200px', border: '1px solid black' }"
+    <va-aspect-ratio
+      :style="{ width: '200px', border: 'var(--border-dashed)' }"
```
Comment (m0ksem) [inline packages/ui/src/components/va-aspect-ratio/VaAspectRatio.stories.ts:13]:
```suggestion
      class="w-48 border-dashed border-2"
```

**Review by m0ksem — CHANGES_REQUESTED** [review]
Would be nice to cover stories with tests.

File: packages/ui/.storybook/storybook-main.scss (hunk 1 above)
Comment (Fsss126) [reply packages/ui/.storybook/storybook-main.scss:9]: I removed tailwind helpers that were in the original stories as it was stated as one of the refactoring steps in #3676

> Try not to use tailwind helpers - we want to remove it from storybook eventually.

@asvae, need your say here on what is the latest agreement on this

Comment (asvae) [reply packages/ui/.storybook/storybook-main.scss:9]: Tailwind helpers are all right. One concern might be not overusing them.

I.e. where it's enough to use only one helper - don't use 3.

And where it's enough to use `<p>` - don't use helpers at all.

Comment (m0ksem) [reply packages/ui/.storybook/storybook-main.scss:9]: I changed guide a bit to
> Please use tailwind for helpers, but not overuse them. Minimum amount of styling is appreciated.

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 3]
```diff
@@ -1,12 +1,386 @@
...
+})
+
+export const AnchorSelector = () => ({
+  components: { VaDropdown },
+  template: `
+      <div
+          style="border: var(--border-dashed)"
+          id="anchor-selector"
+      >
+        Click me
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:225]:
```suggestion
        Anchor
```

Comment (asvae) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:225]: Try to use as little variety as possible.

I.e. anchor should always be a div with "Anchor" text if there is no good reason to use anything else. Should also use the same styling for all cases.

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 4]
```diff
@@ -1,12 +1,386 @@
...
+      </div>
+      <va-dropdown
+          anchor-selector="#anchor-selector"
+      >
+        Anchor selector
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:230]: Here it should be Dropdown content or Content in all cases.

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 5]
```diff
@@ -1,12 +1,386 @@
...
+})
+
+export const InnerAnchorSelector = () => ({
   components: { VaDropdown },
-  template: '<VaDropdown/>',
+  template: `
+      <va-dropdown
+          inner-anchor-selector="#inner-anchor-selector"
+      >
+        <template #anchor>
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:241]: Need a test case showing that you can only trigger via inner selector and can't trigger by clicking on anything else in #anchor slot.

Might make sense to have 2 divs in slot:
[anchor]
[not anchor]

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 6]
```diff
@@ -1,12 +1,386 @@
...
+              Click me
+            </div>
+          </div>
+        </template>
+        Inner anchor selector
+      </va-dropdown>
+    `,
+})
+
+export const AnchorBoundProps = () => ({
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:256]:
```suggestion
export const AnchorSlotProps = () => ({
```

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 7]
```diff
@@ -1,12 +1,386 @@
...
+    `,
+})
+
+export const KeepHeight = () => ({
+  components: { VaDropdown, VaButton, VaDropdownContent },
+  template: `
+      <div style="height: 1000px">
+        <va-dropdown>
+          <template #anchor>
+            <va-button>Anchor</va-button>
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:276]: let's either use va-button everywhere or nowhere

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 8]
```diff
@@ -1,12 +1,386 @@
...
+          <div>
+            <va-icon :name="isOpened ? 'va-arrow-up' : 'va-arrow-down'"/>
+          </div>
+        </template>
+        Clicked
+      </va-dropdown>
+    `,
+})
+
+export const KeepHeight = () => ({
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:270]: Really unclear what this example shows (I have no idea 🤷).

If we can showcase that via interaction or naming - we should.

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 9]
```diff
@@ -1,12 +1,386 @@
...
+              </div>
+            </template>
+            Dropdown
+          </va-dropdown>
+        </div>
+      </scroll-container>
+    `,
+})
+
+export const AutoplacementWithHeight: StoryFn = () => ({
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:319]:
```suggestion
export const AutoplacementWithScroll: StoryFn = () => ({
```

Probably could be both horizontal and vertical

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 10]
```diff
@@ -1,12 +1,386 @@
...
+            <div style="height: 600px; width: 100px;">
+              Hi
+            </div>
+          </va-dropdown-content>
+        </va-dropdown>
+      </div>
+    `,
+})
+
+export const Autoplacement: StoryFn = () => ({
```
Comment (asvae) [inline packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:288]: Need an example for default autoplacement behavior.

It's not clear to me how using `autoplacementTargetRef` differs from not using it.

**Review by asvae — CHANGES_REQUESTED** [review]
Great work!

Left some story suggestions. We can merge after fix!

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 11]
```diff
@@ -1,12 +1,386 @@
...
+      <va-dropdown
+          keepAnchorWidth
+      >
+        <template #anchor>
+          <div>
+            Short anchor
+          </div>
+        </template>
+        <va-dropdown-content>
+          Loooooooong Dropdown
```
[the remark this answers is not in the document]
Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:131]: As I got it, the point was not to test scroll behaviour but to showcase that it matches it in both of these two cases and having both corner cases will help to catch a bug

[the remark this answers is not in the document]
Comment (asvae) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:131]: I would consider 
* modal scales with anchor
* and modal has scroll

to be different cases, doesn't seem likely for me to have an edge case that involves 2 of these.

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts (hunk 10 above)
Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:288]: Default autoplacement behaviour without target doesn't seem to handle overflow properly and dropdown gets positioned outside of scroll parent. Is this needed to be showcased? I thought that in this case target needs to be specified for everything to work correctly and I doubt that improper usage should be shown

Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:288]: Or is current behaviour without specifying the target considered a bug?

File: packages/ui/.storybook/storybook-main.scss (hunk 1 above)
Comment (Fsss126) [reply packages/ui/.storybook/storybook-main.scss]: Returned tailwind helpers 🙌

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 12]
```diff
@@ -1,12 +1,386 @@
...
+      >
+        <template #anchor>
+          Hover me
+        </template>
+        Hovered
+      </va-dropdown>
+    `,
+})
+
+export const PlacementTop = () => ({
```
[the remark this answers is not in the document]
Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts]: Done

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 13]
```diff
@@ -1,12 +1,386 @@
...
+        </va-dropdown-content>
+      </va-dropdown>
+    `,
+})
+
+export const CustomWidth = () => ({
+  components: { VaDropdown, VaValue, VaDropdownContent },
+  template: `
+      <VaValue #default="v" :default-value="true">
+        <button @click="v.value = !v.value">{{ v.value ? 'Short' : 'Long' }}</button>
```
[the remark this answers is not in the document]
Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts]: Changed it

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts [hunk 14]
```diff
@@ -1,12 +1,386 @@
...
+            <va-dropdown-content :style="{ minWidth: getAnchorWidth() }">
+              {{ v.value ? 'Short' : 'Loooong drooooopdown wwiiiiiidth' }}
+            </va-dropdown-content>
+          </template>
+        </va-dropdown>
+      </VaValue>
+    `,
+})
+
+export const Cursor = () => ({
```
[the remark this answers is not in the document]
Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts]: Removed trigger

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts (hunk 6 above)
Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts]: Renamed it

File: packages/ui/src/components/va-dropdown/VaDropdown.stories.ts (hunk 9 above)
Comment (Fsss126) [reply packages/ui/src/components/va-dropdown/VaDropdown.stories.ts:387]: Left it verrical as horizontal scroll is showcased in story demonstrating sticking to edges behaviour

**Review by asvae — APPROVED** [review]
Great work, thanks!
